### PR TITLE
Implemented get_firewall_policies

### PR DESCRIPTION
**Pull request by awlx**
I implemented a new method to get all defined IPv4 Firewall Policies on a Fortigate. 

I hope this PR complies with the guidelines.

I used https://github.com/napalm-automation/napalm-fortios/commit/341e0342c6797a4d8c798dc24e9b5e816ff5dc6e as an orientation as well as the contribution guidelines.

### CODE REVIEW

**Comment by dbarrosop** [comment]
Relates to napalm-automation/napalm-base#136

File: napalm_fortios/fortios.py
Comment (dbarrosop) [inline napalm_fortios/fortios.py]: it's gonna be cleaner if you do

```
my_key = "{}-{}".format(policy[key]['srcintf'], policy[key]['dstintf'])
default_policy[my_key]['position'] = position
default_policy[my_key]['id'] = key
...
```

File: napalm_fortios/fortios.py
Comment (mirceaulinic) [inline napalm_fortios/fortios.py]: You could:

```python
enabled = 'status' in policy[key]
```

File: napalm_fortios/fortios.py
Comment (mirceaulinic) [inline napalm_fortios/fortios.py]: Similarly here:

```python
logtraffic = policy[key]['logtraffic'] if 'logtraffic' in  policy[key] else False
```

File: napalm_fortios/fortios.py
Comment (mirceaulinic) [inline napalm_fortios/fortios.py]: And again:

```python
action = 'permit' if 'action' in policy[key] else 'reject'
```

**Comment by mirceaulinic** [comment]
@awlx I have never used Fortinet, but looking at the config, my understanding is that we could use ```uuid``` as a replacement for policy name. Can you confirm it is unique as the name would suggest?

File: napalm_fortios/fortios.py
Comment (mirceaulinic) [inline napalm_fortios/fortios.py]: I don't understand this line. What's the point?

**Review by mirceaulinic — CHANGES_REQUESTED** [review]
In addition to what David asked, I have suggested few more small changes for more readable code.

**Comment by dbarrosop** [comment]
@mirceaulinic The uuid is unique per ACE so you have multiple on the same ACL. Well... what it would be the equivalent of an ACL : )

File: napalm_fortios/fortios.py
Comment (awlx) [reply napalm_fortios/fortios.py]: I strip the leading white spaces.

File: napalm_fortios/fortios.py
Comment (awlx) [reply napalm_fortios/fortios.py]: The whole key does not exist when the policy is disabled. That's why I only can check if the key is there.

File: napalm_fortios/fortios.py
Comment (awlx) [reply napalm_fortios/fortios.py]: Same here.

File: napalm_fortios/fortios.py
Comment (mirceaulinic) [reply napalm_fortios/fortios.py]: What I proposed does the same thing, but more readable.

File: napalm_fortios/fortios.py
Comment (mirceaulinic) [reply napalm_fortios/fortios.py]: For this purpose, there's no point in using regex. You can use: ``` line.strip()```.

**Comment by dbarrosop** [comment]
Ok, I think this is good to go, let's wait and see if @mirceaulinic confirms his suggestions are addressed and/or if someone else has some comment but I think this is almost ready to go : )

**Comment by mirceaulinic** [comment]
Oh, yes! Sorry, forgot to reply on this. Looks good to me - we only need to release napalm-base 0.20.0 and re-run the tests before merging.

**Review by mirceaulinic — APPROVED** [review]

**Comment by mirceaulinic** [comment]
@awlx have you run the tests locally for this method, using the napalm-base code from https://github.com/napalm-automation/napalm-base/pull/136?

**Comment by awlx** [comment]
Yes :).

```
bash-3.2$ nosetests --with-coverage --cover-package napalm_fortios -v TestFortiOSDriver:TestGetterFortiOSDriver.test_get_firewall_policies
test_get_firewall_policies (TestFortiOSDriver.TestGetterFortiOSDriver) ... ok

----------------------------------------------------------------------
Ran 1 test in 0.001s

OK

**Comment by mirceaulinic** [comment]
@awlx Thanks!

**Comment by dbarrosop** [comment]
@awlx can you rebase? There are some conflicts after another PR that has just been merged.

Thx!

**Comment by dbarrosop** [comment]
As discussed on slack, let's close this one and open a new PR from a new branch : )

**Pull request closed without merging**
